src/ompl/base/objectives/src/PathLengthOptimizationObjective.cpp

```diff
@@ -62,10 +64,14 @@ ompl::base::Cost ompl::base::PathLengthOptimizationObjective::motionCostHeuristi
     return motionCost(s1, s2);
 }
 
-#if OMPL_HAVE_EIGEN3
 ompl::base::InformedSamplerPtr ompl::base::PathLengthOptimizationObjective::allocInformedStateSampler(const ProblemDefinitionPtr probDefn, unsigned int maxNumberCalls) const
 {
-    // Make the direct path-length informed sampler and return
+    // Make the direct path-length informed sampler and return. If OMPL was compiled with Eigen, a direct version is available, if not a rejection-based technique can be used
+#if OMPL_HAVE_EIGEN3
     return boost::make_shared<PathLengthDirectInfSampler>(probDefn, maxNumberCalls);
-}
+#else
+    OMPL_ERROR("%s: Direct sampling of the path-length objective requires Eigen, but OMPL was compiled without Eigen support. If possible, please install Eigen and recompile OMPL. If this is not possible, you can manually create an instantiation of RejectionInfSampler to approximate the behaviour of direct informed sampling. ", description_.c_str());
+    // Return a null pointer to avoid compiler warnings
+    return ompl::base::InformedSamplerPtr();
```

Comment:
* I'd say throwing an exception is better, since nothing good will happen after the null pointer is returned.
 * A compile time error would be nicer, but a runtime error is fine.